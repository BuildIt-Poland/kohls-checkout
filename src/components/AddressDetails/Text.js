import styled from 'styled-components';

import { default as DefaultText } from '../Text';

const Text = styled(DefaultText)`
  margin: 0;
`;

export default Text;
