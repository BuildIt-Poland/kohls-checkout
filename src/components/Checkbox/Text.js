import styled from 'styled-components';

import { fontWeightNormal, spacingSmall, fontMedium, colorBlack } from '../../styles/designTokens';

const Text = styled.span`
  display: block;
  color: ${colorBlack};
  font-size: ${fontMedium};
  font-weight: ${fontWeightNormal};
  margin: ${spacingSmall};
  cursor: pointer;
`;

export default Text;
