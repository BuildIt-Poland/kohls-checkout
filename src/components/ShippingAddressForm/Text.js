import styled from 'styled-components';

import { colorDarkGrey, spacingMedium } from '../../styles/designTokens';
import TextBase from '../Text';

const Text = styled(TextBase)`
  color: ${colorDarkGrey};
  margin-bottom: ${spacingMedium};
`;

export default Text;
