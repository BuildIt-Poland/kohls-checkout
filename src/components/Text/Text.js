import styled from 'styled-components';
import PropTypes from 'prop-types';

import {
  fontMedium,
  fontWeightNormal,
  fontWeightBold,
  spacingSmall,
  colorBlack,
  colorDarkGrey,
  colorTurquoiseDimmed
} from '../../styles/designTokens';

export function fontColor({ highlight, dimmed }) {
  if (highlight) {
    return colorTurquoiseDimmed;
  } else if (dimmed) {
    return colorDarkGrey;
  }
  return colorBlack;
}

const Text = styled.p`
  font-size: ${fontMedium};
  text-decoration: ${({ underline }) => (underline ? 'underline' : 'none')};
  font-weight: ${({ bold }) => (bold ? fontWeightBold : fontWeightNormal)};
  line-height: 1.4;
  margin: 0;
  margin-bottom: ${spacingSmall};
  color: ${fontColor};

  &:last-child {
    margin-bottom: 0;
  }
`;

Text.propTypes = {
  /** Highlight text */
  highlight: PropTypes.bool,
  /** Dimmed text */
  dimmed: PropTypes.bool,
  /** Underline text */
  underline: PropTypes.bool,
  /** Bold text */
  bold: PropTypes.bool
};

export default Text;
