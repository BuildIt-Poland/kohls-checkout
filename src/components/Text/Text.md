```js
<Text>Basic Text</Text>
<Text dimmed>Dimmed Text</Text>
<Text bold>Bold Text</Text>
<Text underline>Underline Text</Text>
<Text highlight>Highlight Text</Text>
<Text highlight bold>Bold hihglight Text</Text>
```
