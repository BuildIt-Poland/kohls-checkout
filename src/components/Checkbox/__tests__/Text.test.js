import React from 'react';
import { create } from 'react-test-renderer';

import Text from '../Text';

describe('COMPONENT - Text', () => {
  it('renders Text component', () => {
    const component = create(<Text />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
