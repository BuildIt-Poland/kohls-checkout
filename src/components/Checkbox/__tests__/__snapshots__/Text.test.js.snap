// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Text renders Text component 1`] = `
.c0 {
  display: block;
  color: #222222;
  font-size: 1.8rem;
  font-weight: 400;
  margin: 1rem;
  cursor: pointer;
}

<span
  className="c0"
/>
`;
